### PR TITLE
Alom Refactoring

### PR DESCRIPTION
**Pull request by eldelentes**
## What does this PR do?

- Commenting all the files.
- Deletes _mediaqueries.scss (Few mixins are now included in _responsive.scss
- Deletes some properties (like user-select) who makes buggy websites on desktop
- Deletes silen clearfix (Is duplicated)
- Rethinking the text align utilities.
- Adds displays utilities
- Rethinking the align utilities and adds flexbox utilities.
- Adds disabled and hidden utilities
- Adds percent calculation and px to rems functions
- Adds _variables.scss files.
- Adds assets pipeline function
- Adds full hieght and fixed rows helpers to grid.
- Adds flexbox grid system
- Adds hide on "device" and hide-only "device"

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 node_modules/
 *.DS_store
 *.swp
+docs/
```
Comment (kurenn) [inline .gitignore:4]: Why ignoring docs? it is being use to publish the website

File: scss/_functions.scss [hunk 2]
```diff
@@ -1,8 +1,87 @@
...
+    }
+
+  }
+}
+
+//############################################//
+//####### Get previus value on maps  ########//
+//###########################################//
+
+@function prev($map, $key, $fallback: false, $return: value) {
```
Comment (kurenn) [inline scss/_functions.scss:70]: Is this for Google maps?

File: scss/_grid.scss [hunk 3]
```diff
@@ -46,29 +24,65 @@ $grid-max-width: 768px !default;
 }
 
 .row {
-  @extend %clearfix;
+  @include clearfix;
```
Comment (kurenn) [inline scss/_grid.scss]: Why change it to mixin, instead of a placeholder?

File: scss/_grid.scss [hunk 4]
```diff
@@ -46,29 +24,65 @@ $grid-max-width: 768px !default;
...
+    }
   }
+}
+
+
+//############################################//
+//########### Flex Grid System ##############//
+//###########################################//
+
+.row-flex {
```
Comment (kurenn) [inline scss/_grid.scss:59]: 👍

File: scss/_reset.scss [hunk 5]
```diff
@@ -10,6 +13,14 @@
...
 
+html {
+	-moz-osx-font-smoothing: grayscale;
+	-webkit-font-smoothing: antialiased;
+	text-rendering: optimizeLegibility;
+	overflow-x: hidden;
+  overflow-y: scroll;
+}
+
 html, body, div, span, applet, object, iframe,
```
Comment (kurenn) [inline scss/_reset.scss:24]: This should be separated with new lines

File: scss/_responsive.scss [hunk 6]
```diff
@@ -1,272 +1,46 @@
...
-  }
+//############################################//
+//####### Hide elements on devices  #########//
+//###########################################//
 
-  &.visible-lg {
-    @media  #{$large-screen-media} {
-      @include responsive-visibility();
+@each $bp in map-keys($breakpoints) {
+  .hide-#{$bp} {
```
Comment (kurenn) [inline scss/_responsive.scss]: Shouldn't this be better with `.hide-on-#{$bp}`?

**Comment by kurenn** [comment]
I don't think the documentation for using the breakpoints is clear enough, it really doesn't say anything, even about the variables, `m`, `l`, there is no place where I can find its value if not the code...

**Comment by kurenn** [comment]
Not really sure about the display utility classes, are they actually being used or necessary?

**Comment by scarlettgarza** [comment]
@kurenn it does says what s, m, l means in the responsive section on the table, but I added the measures on the media queries docs too.

**Comment by kurenn** [comment]
That's good enough for now! Thanks 👍

**Pull request merged**